### PR TITLE
QueueMovies Endpoint

### LINKED ISSUE #9: Movies - queueMovies endpoint
Endpoint details:- https://api.helioviewer.org/docs/v2/api/api_groups/movies.html#queuemovie

### PR DESCRIPTION
**Pull request by akash5100**
* closes #9

### CODE REVIEW

File: hvpy/api_groups/movies/queue_movie.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+
+class queueMovieInputParameters(HvpyParameters):
+    """
+    Handles the input parameters of the queueMovie API.
+
+    {Shared}
+    Attributes
+    ----------
+    startTime : datetime
```
Comment (nabobalis) [inline hvpy/api_groups/movies/queue_movie.py:17]:
```suggestion
    startTime : datetime.datetime
```

File: hvpy/api_groups/movies/queue_movie.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+class queueMovieInputParameters(HvpyParameters):
+    """
+    Handles the input parameters of the queueMovie API.
+
+    Attributes
+    ----------
+    {Shared}
+    startTime : datetime
+        Date and time of the first frame of the movie.
+    endTime : datetime
```
Comment (nabobalis) [inline hvpy/api_groups/movies/queue_movie.py:19]:
```suggestion
    endTime : datetime.datetime
```

File: hvpy/api_groups/movies/queue_movie.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    x1: Optional[str] = None
+    y1: Optional[str] = None
+    x2: Optional[str] = None
+    y2: Optional[str] = None
+    callback: Optional[str] = None
+    size: Optional[int] = None
+    movieIcons: Optional[int] = None
+    followViewport: Optional[int] = None
+    reqObservationDate: Optional[datetime] = None
+    _date_validator = validator("startTime", "endTime", "reqObservationalDate", allow_reuse=True, check_fields=False)(
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply hvpy/api_groups/movies/queue_movie.py:109]: You should not need that. There must be a typo somewhere.

**Comment by akash5100** [comment]
> Can this be rebased as well?

We should leave this PR draft until we close the BASEURL issue. After solving that we can rebase it with main.

**Comment by nabobalis** [comment]
@akash5100 

> > Can this be rebased as well?
> 
> We should leave this PR draft until we close the BASEURL issue. After solving that we can rebase it with main.

Just because this is a draft PR, does not mean we shouldn't rebase this.

**Comment by nabobalis** [comment]
> Oops! I did a rebase but I think I messed up something. The files I didn't changed is showing up in files changed.

Do it again with an updated remote? YOu might need to drop some commits now tho.

**Comment by akash5100** [comment]
> > Oops! I did a rebase but I think I messed up something. The files I didn't changed is showing up in files changed.
> 
> You want to make sure what you doing a rebase on is up to date using `git remote update`, you will want to try `git rebase -i upstream/main`, if you see extra commits that aren't part of the original PR, you will need to drop them manually

Thanks, can you make a review?

File: hvpy/api_groups/movies/queue_movie.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    y2
+        The vertical offset of the bottom-right corner of the field of view with respect to the center of the Sun (in arcseconds).
+        Used in conjunction with `x1`, `y1`, and `x2`.
+        Default value is `None`, optional.
+    callback
+        Wrap the response object in a function call of your choosing.
+        Default value is `None`, optional.
+    size
+        Scale video to preset size.
+        Default value is `0`, optional.
```
Comment (nabobalis) [inline hvpy/api_groups/movies/queue_movie.py:89]:
```suggestion
        Default value is 0, optional.
```
I don't think a value needs to be quoted like this.

File: hvpy/api_groups/movies/queue_movie.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    eventsLabels
+        Boolean to indicate if the event labels should be included in the movie.
+    imageScale
+        Image scale in arcseconds per pixel.
+    format
+        Movie format (mp4, webm, flv).
+        Default value is `mp4`, optional.
+    frameRate
+        Movie frames per second.
+        Default value is `15` frames per second, optional.
```
Comment (nabobalis) [inline hvpy/api_groups/movies/queue_movie.py:34]:
```suggestion
        Default value is 15 frames per second, optional.
```

File: hvpy/api_groups/movies/queue_movie.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+        Image datasource layers to include in the movie.
+    events
+        List of feature/event types and FRMs to use to annotate the movie.
+    eventsLabels
+        Boolean to indicate if the event labels should be included in the movie.
+    imageScale
+        Image scale in arcseconds per pixel.
+    format
+        Movie format (mp4, webm, flv).
+        Default value is `mp4`, optional.
```
Comment (nabobalis) [inline hvpy/api_groups/movies/queue_movie.py:31]:
```suggestion
        Default value is "mp4", optional.
```

File: hvpy/facade.py [hunk 7]
```diff
@@ -298,3 +299,86 @@ def downloadScreenshot(id: int) -> Union[bytes, str, Dict[str, Any]]:
...
+    >>> from hvpy import queueMovie
+    >>> queueMovie(
+    ...     startTime=datetime(2022, 7, 21, 12, 12, 12),
+    ...     endTime=datetime(2022, 7, 22, 12, 12, 12),
+    ...     layers="[12,7,22],[13,7,11]",
+    ...     events="[AR,HMI_HARP;SPoCA,1],[CH,all,1]",
+    ...     eventsLabels=False,
+    ...     imageScale=2.44,
+    ... )
+    {'id': ...}
```
Comment (nabobalis) [inline hvpy/facade.py:352]: What does the full response look like out of curiosity?

File: hvpy/utils.py [hunk 8]
```diff
@@ -15,11 +15,13 @@ def decorator(func):
...
 
-def convert_date_to_isoformat(v: datetime) -> str:
+def convert_date_to_isoformat(v: datetime) -> Union[str, None]:
     """
     Converts the date from a datetime object to a string in the ISO format.
     """
-    return v.isoformat() + "Z"
+    if v:
+        return v.isoformat() + "Z"
+    return None
```
Comment (nabobalis) [inline hvpy/utils.py:24]: What happened here?

File: hvpy/facade.py (hunk 7 above)
Comment (akash5100) [reply hvpy/facade.py:352]:
```
{
   'id': 'gxRN5', 
   'eta': 274, 
   'queue': 0, 
   'token': '5d537e2d54dbdc725cbb26dda830e2f0'
}
```

File: hvpy/utils.py (hunk 8 above)
Comment (akash5100) [reply hvpy/utils.py:24]: The parameter `reqObservationDate` is Optional DateTime, so when we pass `None` - it throws an error saying, None has no attribute called `isoformat`.

Comment (nabobalis) [reply hvpy/utils.py:24]: Makes sense.

Comment (nabobalis) [inline hvpy/utils.py:24]:
```suggestion
    if v is not None:
        return v.isoformat() + "Z"
```
What happens if this is the code?

Comment (akash5100) [reply hvpy/utils.py:24]: Yes, same output but better code.

**Review by dgarciabriseno — APPROVED** [review]

**Review by nabobalis — APPROVED** [review]

**Pull request merged**